compiler/src/dotty/tools/dotc/util/MurmurLongHash3.scala

```diff
@@ -0,0 +1,261 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___     Scala API                            **
+**    / __/ __// _ | / /  / _ |    (c) 2003-2013, LAMP/EPFL             **
+**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/               **
+** /____/\___/_/ |_/____/_/ | |                                         **
+**                          |/                                          **
+\*                                                                      */
+
+/* Copied and adapted from scala.util.hashing.MurmurHash3 to use Longs */
+
+package dotty.tools.dotc.util
+
+import java.lang.Long.{ rotateLeft => rotl }
+
+private[util] class MurmurLongHash3 {
```

Comment:
Good idea. But I need to reimplement the algorithm from the source. @DarkDimius showed me some issues with my simple port. I will try to make it compatible `MurmurHash3` to be able to just add it inside in the future.